### PR TITLE
Run scala orb as non-root sbtuser

### PR DESCRIPTION
**Pull request by fpopic**
We install code-climate in /tmp folder (the easiest and we are sure every non-root user has access)

### CODE REVIEW

File: src/code-climate/commands/install.yaml [hunk 1]
```diff
@@ -7,15 +7,7 @@ parameters:
...
-          if which sudo > /dev/null; then
-            sudo curl -s -o /usr/local/bin/cc-test-reporter -L https://codeclimate.com/downloads/test-reporter/test-reporter-<<parameters.version>>-linux-amd64
-            sudo chmod +x /usr/local/bin/cc-test-reporter
-          else
-            curl -s -o /usr/local/bin/cc-test-reporter -L https://codeclimate.com/downloads/test-reporter/test-reporter-<<parameters.version>>-linux-amd64
-            chmod +x /usr/local/bin/cc-test-reporter
-          fi
-        else
-          echo "Test reporter is already installed."
-        fi
```
Comment (fpopic) [inline src/code-climate/commands/install.yaml:20]: @travelaudience/core-services our CI/orbs use root user as well, the issue with using non-root is to find a common directory that can be used by nonroot user (needs to be able to write and chmod files).

Scala orb base image gives nonroot user `sbtuser` [ /opt](https://github.com/hseeberger/scala-sbt/blob/master/debian/Dockerfile#L53) folder ownership.

I think the target destination of code-climate binaries should be configurable, but the issue is that most likely this folder won't be on $PATH, and export of new PATH variable is only per shell/run cmd in circcleci, so maybe better to use just `/tmp`?

File: src/scala/executors/scala.yaml [hunk 2]
```diff
@@ -15,5 +15,6 @@ parameters:
     default: "~/repo"
 docker:
   - image: hseeberger/scala-sbt:<< parameters.version >>
+    user: sbtuser
```
Comment (fpopic) [inline src/scala/executors/scala.yaml:18]: setting nonroot sbtuser as USER

File: src/code-climate/commands/install.yaml (hunk 1 above)
Comment (fpopic) [reply src/code-climate/commands/install.yaml:20]: For scala /tmp works https://app.circleci.com/pipelines/github/travelaudience/workflow-dw-destinations/1415/workflows/e41162fb-8afb-41c0-a240-a1c826100f7c/jobs/6211

File: .circleci/config.yml [hunk 3]
```diff
@@ -219,6 +220,7 @@ workflows:
           requires:
             - python-lint
             - python-pack
+            - code-climate-dev-publish
```
Comment (fpopic) [inline .circleci/config.yml:223]: when orb depends on another orb the CI dependencies should reflect that so you can modify them in one PR

File: src/code-climate/commands/install.yaml (hunk 1 above)
Comment (fpopic) [reply src/code-climate/commands/install.yaml:20]: does this work 

```bash
echo 'export PATH=/path/to/code-climate/bin:$PATH.' >> $BASH_ENV
```

across different commands within orb job. Can try maybe next week.

File: src/code-climate/commands/format_coverage.yaml [hunk 4]
```diff
@@ -24,7 +24,10 @@ parameters:
     default: coverage/codeclimate.json
     description: The output path. (default "coverage/codeclimate.json")
     type: string
+  cc_test_reporter_path:
+    default: /usr/local/bin
+    description: Path where Code Climate test reporter binary is installed
 steps:
   - run:
-      command: cc-test-reporter format-coverage <<# parameters.additional_flags>> "<<parameters.additional_flags>>" <</ parameters.additional_flags>> "<<parameters.coverage_file>>" --input-type <<parameters.input_type>> --output "<<parameters.output>>"
+      command: <<params.cc_test_reporter_path>>/cc-test-reporter format-coverage <<# parameters.additional_flags>> "<<parameters.additional_flags>>" <</ parameters.additional_flags>> "<<parameters.coverage_file>>" --input-type <<parameters.input_type>> --output "<<parameters.output>>"
```
Comment (fpopic) [inline src/code-climate/commands/format_coverage.yaml:32]: so I just added parameter and put the current value as default so it doesn't break other users

File: src/scala/@orb.yml [hunk 5]
```diff
@@ -5,4 +5,4 @@ description: |
   see github page for more info: https://github.com/travelaudience/orbs
 
 orbs:
-  code-climate: travelaudience/code-climate@0.1
+  code-climate: travelaudience/code-climate@dev:code_climate_parametrize
```
Comment (fpopic) [inline src/scala/@orb.yml:8]:
```suggestion
  code-climate: travelaudience/code-climate@0.1
```
```suggestion
  code-climate: travelaudience/code-climate@dev:code_climate_parametrize
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -199,6 +199,7 @@ workflows:
           requires:
             - scala-lint
             - scala-pack
+            - code-climate-dev-publish
```
Comment (fpopic) [inline .circleci/config.yml:202]:
```suggestion
```

**Review by nvankaam — DISMISSED** [review]

File: src/scala/@orb.yml [hunk 7]
```diff
@@ -5,4 +5,4 @@ description: |
   see github page for more info: https://github.com/travelaudience/orbs
 
 orbs:
-  code-climate: travelaudience/code-climate@0.1
+  code-climate: travelaudience/code-climate@dev:run_scala_orb_containers_as_nonroot_sbtuser
```
Comment (bpietraga) [inline src/scala/@orb.yml]:
```suggestion
  code-climate: travelaudience/code-climate@0.1.3
```

File: src/scala/@orb.yml [hunk 8]
```diff
@@ -5,4 +5,4 @@ description: |
   see github page for more info: https://github.com/travelaudience/orbs
 
 orbs:
-  code-climate: travelaudience/code-climate@0.1
+  code-climate: travelaudience/code-climate@0.1.3
```
Comment (bpietraga) [inline src/scala/@orb.yml:8]:
```suggestion
  code-climate: travelaudience/code-climate@0.1
```

**Review by bpietraga — APPROVED** [review]
LGTM

**Pull request merged**